onnxruntime/core/framework/session_state.h

```diff
@@ -59,6 +60,32 @@ struct MemoryPatternGroup;
 class MemoryInfo;
 #endif
 
+#ifdef ENABLE_TRAINING
+struct PartialGraphExecutionState {
+ public:
+  PartialGraphExecutionState() {
+    execution_frame_ = nullptr;
+  }
+
+  ~PartialGraphExecutionState() {
+    execution_frame_.~unique_ptr<ExecutionFrame>();
```

Comment:
This looks unusual to me... is there some need to invoke the destructor like this instead of allowing the execution frame to be destroyed implicitly (or by calling `reset()` if there's some need to destroy the pointed-to execution frame earlier)?

(If there is a need, please add a short comment explaining why.)

Reply:
Had it to debug stuff but we should just be using default.

---
In reply to: [613561040](https://github.com/microsoft/onnxruntime/pull/7324#discussion_r613561040) [](ancestors = 613561040)